### PR TITLE
retry creating the browser on WebDriverException

### PR DESCRIPTION
**Pull request by gabriel-v**
about 3 times

closes https://github.com/liquidinvestigations/liquidinvestigations/issues/240

### CODE REVIEW

File: tests/testsuite/test_acceptance_with_browsers.py [hunk 1]
```diff
@@ -121,10 +122,20 @@ def wait_for_reconfigure():
...
+                                  wait_time=15,
+                                  **BROWSER_OPTS[browser_name]) as browser:
+                browser.driver.set_window_size(1920, 1080)
+                browser.visit(URL)
+                yield browser
+                break
+        except WebDriverException:
+            continue
+    else:
+        pytest.fail("Got WebDriverException 3 times in a row")
```
Comment (mgax) [inline tests/testsuite/test_acceptance_with_browsers.py]: I think the 3rd time we should re-raise the exception, and i'm really not a fan of the indentation on line 127. How about this?

```python
    retry = 3
    while retry:
        retry -= 1
        try:
            browser = splinter.Browser(
                browser_name,
                headless=True,
                wait_time=15,
                **BROWSER_OPTS[browser_name]
            )
            browser.driver.set_window_size(1920, 1080)
            browser.visit(URL)
            yield browser
            return
        except WebDriverException:
            if retry:
                continue
            else:
                raise
```

Comment (gabriel-v) [reply tests/testsuite/test_acceptance_with_browsers.py]: we still need the `with` there, so it calls `close` on the browser driver

File: tests/testsuite/test_acceptance_with_browsers.py [hunk 2]
```diff
@@ -121,10 +122,25 @@ def wait_for_reconfigure():
...
+                        browser_name,
+                        headless=True,
+                        wait_time=15,
+                        **BROWSER_OPTS[browser_name],
+                    ) as browser:
+                browser.driver.set_window_size(1920, 1080)
+                browser.visit(URL)
+                yield browser
+                return
+        except WebDriverException:
```
Comment (mgax) [inline tests/testsuite/test_acceptance_with_browsers.py]: I'd move the `try/except` to only cover the two calls to the browser (`set_window_size` and `visit`). Then move `yield` and `return` to an `else` clause for the `try/except`. The goal is to narrow the `try/except` to only the lines where we expect a failure.

Comment (gabriel-v) [reply tests/testsuite/test_acceptance_with_browsers.py]: `splinter.Browser(...)` is what throws the `WebDriverException`

Comment (gabriel-v) [reply tests/testsuite/test_acceptance_with_browsers.py]: i'll put the `try... except` around the `splinter.Browser(...)` and stop using `with`

**Comment by gabriel-v** [comment]
i'll just retry the build if it fails this way (for now)

**Pull request closed without merging**